### PR TITLE
RavenDB-19726 Microsoft logs

### PR DESCRIPTION
**Pull request by haludi**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19726

### Additional description
The logs will be written to the main log source.
By default, log configuration is disabled.

### Type of change
- New feature

### How risky is the change?
- Moderate 

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.

### Testing by Contributor
- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team
- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- It requires further work in the Studio

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 1]
```diff
@@ -186,10 +188,30 @@ public async Task SetMicrosoftConfiguration()
...
+                var provider = Server.GetService<MicrosoftLoggingProvider>();
+                await provider.Configuration.ReadConfigurationAsync(RequestBodyStream(), context, reset);
+                provider.ApplyConfiguration();
             }
 
             NoContentStatus();
         }
+        
+        [RavenAction("/admin/logs/microsoft/enable", "POST", AuthorizationStatus.Operator)]
+        public async Task EnableMicrosoftLog()
```
Comment (grisha-kotler) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs:200]: Can you fix the warning?

Comment (grisha-kotler) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Can you fix the warning?

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 2]
```diff
@@ -186,10 +188,30 @@ public async Task SetMicrosoftConfiguration()
...
+        {
+            var provider = Server.GetService<MicrosoftLoggingProvider>();
+            provider.ApplyConfiguration();
+
+            NoContentStatus();
+        }
+        
+        [RavenAction("/admin/logs/microsoft/disable", "POST", AuthorizationStatus.Operator)]
+        public async Task DisableMicrosoftLog()
```
Comment (grisha-kotler) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Same here

File: src/Raven.Server/Utils/MicrosoftLogging/MicrosoftLoggingConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    {
+        FileStream stream;
+        try
+        {
+            stream = File.Open(configurationPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+        }
+        catch (Exception e)
+        {
+            if (e is FileNotFoundException)
+                return ;
```
Comment (grisha-kotler) [inline src/Raven.Server/Utils/MicrosoftLogging/MicrosoftLoggingConfiguration.cs]: Maybe dismiss the notification here as well?

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 1 above)
Comment (haludi) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Done

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 2 above)
Comment (haludi) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Done

File: src/Raven.Server/Utils/MicrosoftLogging/MicrosoftLoggingConfiguration.cs (hunk 3 above)
Comment (haludi) [reply src/Raven.Server/Utils/MicrosoftLogging/MicrosoftLoggingConfiguration.cs]: Done

File: src/Raven.Server/Config/Categories/LogsConfiguration.cs [hunk 4]
```diff
@@ -62,40 +55,11 @@ public class LogsConfiguration : ConfigurationCategory
         [ConfigurationEntry("Logs.Microsoft.Disable", ConfigurationEntryScope.ServerWideOnly)]
         public bool DisableMicrosoftLogs { get; set; }
```
Comment (ppekrol) [inline src/Raven.Server/Config/Categories/LogsConfiguration.cs:56]: Shouldn't this be set to 'false' with another configuration option of log level set to None?

File: src/Raven.Server/Utils/MicrosoftLogging/LogLeveToString.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+using Microsoft.Extensions.Logging;
+
+namespace Raven.Server.Utils.MicrosoftLogging;
+
+public static class LogLeveToString
```
Comment (ppekrol) [inline src/Raven.Server/Utils/MicrosoftLogging/LogLeveToString.cs]: typo

**Review by gregolsky — APPROVED** [review]

File: src/Raven.Server/Config/Categories/LogsConfiguration.cs (hunk 4 above)
Comment (haludi) [reply src/Raven.Server/Config/Categories/LogsConfiguration.cs:56]: The disable here does nothing except to not call the `ApllyConfiguration` and ensure all future created logger will be set to None.
* In my observation all loggers are created before but just to be sure.

If no configuration is applied the default is `None`.
This is also the same for empty or a not exit config file.

File: src/Raven.Server/Utils/MicrosoftLogging/LogLeveToString.cs (hunk 5 above)
Comment (haludi) [reply src/Raven.Server/Utils/MicrosoftLogging/LogLeveToString.cs]: Done

**Pull request merged**